climakitae/explore/vulnerability.py

```diff
@@ -0,0 +1,338 @@
+"""Tools for CAVA vulnerability assessment pilot"""
+
+from climakitae.util.utils import get_closest_gridcell
+from climakitae.core.data_load import load
+from climakitae.core.data_interface import Select
+from climakitae.core.data_export import export
+import xarray as xr
+import numpy as np
+import warnings
+
+# Ignore specific warnings about division errors to create more elegant logging
+warnings.filterwarnings("ignore", message="invalid value encountered in sqrt")
+warnings.filterwarnings("ignore", message="divide by zero encountered in divide")
+
+
+def metric_agg(da, metric, threshold):
+    """
+    Calculate a metric on the given DataArray (da) using the specified metric type
+    and threshold.
+
+    Parameters:
+    -----------
+    da : xarray.DataArray
+        Input DataArray to compute metric and treshold on.
+    metric : str
+        The metric type to calculate. Supported options are "min", "max",
+        "mean", and "median".
+    threshold : float
+        The threshold value for comparison.
+
+    Returns:
+    --------
+    metric_data : xarray.DataArray
+        Resulting DataArray from the calculated metric and threshold applied.
+    """
+    metric_map = {
+        "min": np.min,
+        "max": np.max,
+        "mean": np.mean,
+        "median": np.median,
+    }
+
+    metric_data = (
+        (da.resample(time="1D").reduce(metric_map[metric]) > threshold)
+        .resample(time="1Y")
+        .sum()
+    )
+
+    return metric_data
+
+
+def check_and_set_params(
```

Comment:
May consider using Pydantic in the next iteration to make validation code more concise.